pandas/core/groupby/generic.py

```diff
@@ -1664,3 +1655,36 @@ def _normalize_keyword_aggregation(kwargs):
         order.append((column,
                       com.get_callable_name(aggfunc) or aggfunc))
     return aggspec, columns, order
+
+
+def _recast_datetimelike_result(result):
```

Comment:
so i would put this in pandas.core.dtypes.cast, our dumping ground for casting, right after / before maybe_infer_to-datetimelike (and you can de-privatize)

Reply:
This is really kludgy code (and is replacing equally kludgy code; kind of like turtles all the way down).  I'd rather keep it close to its only usage and hope it is ripped out by someone who knows this part of the code better

Reply:
as i said this look a *lot* like maybe_convert_objects, try to use that here

Reply:
Note also: we only have 5 tests that go through this path